elastica/boundary_conditions.py

```diff
@@ -266,11 +273,11 @@ class GeneralConstraint(ConstraintBase):
 
     def __init__(
         self,
-        *fixed_data,
-        translational_constraint_selector: Optional[np.ndarray] = None,
-        rotational_constraint_selector: Optional[np.array] = None,
-        **kwargs,
-    ):
+        *fixed_data: Any,
+        translational_constraint_selector: Optional[NDArray[np.bool_]] = None,
+        rotational_constraint_selector: Optional[NDArray[np.bool_]] = None,
```

Comment:
`np.bool` doesn't exist.
```sh
AttributeError: module 'numpy' has no attribute 'bool'.
`np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
    https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations.
```